### PR TITLE
qa_crowbarsetup.sh: prepare HA controller node before reboot (SOC-10960)

### PR DESCRIPTION
**Pull request by stefannica**
The `onadmin_rebootcloud` logic from qa_crowbarsetup.sh is not
handling rebooting a cloud node correctly when it is part of
an HA setup. According to the official SLE-HA docs [1], the
controller node needs to be put into standby mode, which
effectively moves all crm resources away from the node and
the pacemaker service has to be stopped before the node can
be safely rebooted.

Not following the correct reboot procedure leads to the node
being fenced half-way through the reboot process, which in turn
causes other problems, such as corrupted root partitions (because
we also have disabled ext journaling on the ECP CI images, to
improve IO performance).

[1] https://documentation.suse.com/sle-ha/15-SP1/single-html/SLE-HA-guide/#pro-ha-maint-shutdown-node

### CODE REVIEW

**Comment by stefannica** [comment]
Also testing with this SOC7 job: https://ci.suse.de/blue/organizations/jenkins/openstack-crowbar/detail/openstack-crowbar/752/pipeline

**Comment by stefannica** [comment]
> LGTM
> 
> How did we make it thus far without this change?

@nicolasbock I don't think this matters too much in mkcloud jobs, because pacemaker can survive an unexpected node shutdown. It's only in the ECP jobs that this causes a real problem, like corrupted root partitions, because we disabled ext journaling in those jobs.

**Comment by stefannica** [comment]
> > LGTM
> > How did we make it thus far without this change?
> 
> @nicolasbock I don't think this matters too much in mkcloud jobs, because pacemaker can survive an unexpected node shutdown. It's only in the ECP jobs that this causes a real problem, like corrupted root partitions, because we disabled ext journaling in those jobs.

Aside from getting the gating jobs to be more reliable, we should aim to test the product in the CI as close as possible to how customers use it, and this change brings us closer to that goal.

**Comment by nicolasbock** [comment]
Totally agree :smile:

File: scripts/qa_crowbarsetup.sh [hunk 1]
```diff
@@ -5583,9 +5583,14 @@ function reboot_controller_clusters
             wait_for 400 5 \
                 "ssh $m_hostname 'if \`which drbdadm &> /dev/null\`; then drbd-overview; ! drbdadm dstate all | grep -v UpToDate/UpToDate | grep -q .; fi'" \
                 "drbd devices to be consistent on node $m_hostname"
+            ssh $m_hostname crm --wait node standby
+            wait_for 30 5 "ssh $m_hostname crm status | grep -q 'Node $m_hostname: standby'" "node $m_hostname to enter crm standby mode"
```
Comment (jsuchome) [inline scripts/qa_crowbarsetup.sh]: Are you sure `$m_hostname` is properly evaluated in the grep command?

Comment (jsuchome) [inline scripts/qa_crowbarsetup.sh]: BTW, you can also use `crm node show $m_hostname | grep standby=on` ... but it probably does not have any advantages over `crm status`

Comment (JanZerebecki) [reply scripts/qa_crowbarsetup.sh]: Yes, single quotes inside double quotes are literals.

**Pull request merged**